container/openejb-core/src/main/java/org/apache/openejb/junit/jupiter/ApplicationComposerExtension.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.openejb.junit.jupiter;
+
+import org.apache.openejb.OpenEJBRuntimeException;
+import org.apache.openejb.testing.ApplicationComposers;
+import org.apache.openejb.testing.SingleApplicationComposerBase;
+import org.junit.jupiter.api.extension.*;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class ApplicationComposerExtension extends ApplicationComposerExtensionBase implements BeforeAllCallback, AfterAllCallback, BeforeEachCallback, AfterEachCallback {
+
+    private static final ExtensionContext.Namespace NAMESPACE = ExtensionContext.Namespace.create(ApplicationComposerExtension.class.getName());
+    private static final SingleApplicationComposerBase BASE = new SingleApplicationComposerBase();
+
+    @Override
+    public void beforeAll(ExtensionContext context) throws Exception {
+
+        if (isPerJvm(context)) {
+            BASE.start(context.getTestClass().orElse(null));
```

Comment:
do you want to check there is no @Module/@Classes/... or so in the class? it would be a misuse of the API

Reply:
Couldn't find any documentation related to `@Application` at all. It seems, it isn't quite often used in the code base nor in the examples? 

`ApplicationComposers#validate()`will complain and fail to start any container, if there is no `@Classes` or `@Default` annotation present?

Probably, I don't get the point here?

Reply:
Okay. I think I am understanding it now. Found some information about `-Dtomee.application-composer.application`, so I guess, we should do the validation.

Reply:
If I get it right, the PER_JVM case takes a fully assembled `@Application` defined in a class specified via `tomee.application-composer.application`. The rational is now to check for additional annotations (modules, classes, ...), which would lead to undefined behaviour. So it requires a check / validation for the following annotations

```
org.apache.openejb.testing.Component; //??
org.apache.openejb.testing.Module;
org.apache.openejb.testing.Classes;
org.apache.openejb.testing.Default //??
org.apache.openejb.testing.Jars; // ??
//similar @Deprecated annotated annotations in *.junit.*
```
on the test class and its related methods? Anything additional, @rmannibucau ?

Reply:
guess you can skip @Deprecated ones, they shouldnt be used in junit5 module anyway (new code with deprecated API sounds fishy from the IDE ;)).

Reply:
Alright - just added a related validation.